src/hotspot/os/linux/osContainer_linux.cpp

```diff
@@ -60,11 +61,18 @@ void OSContainer::init() {
   if (cgroup_subsystem == NULL) {
     return; // Required subsystem files not found or other error
   }
-  // We need to update the amount of physical memory now that
-  // cgroup subsystem files have been processed.
-  if ((mem_limit = cgroup_subsystem->memory_limit_in_bytes()) > 0) {
+  if ((mem_limit = cgroup_subsystem->memory_limit_in_bytes()) > 0 &&
+       mem_limit < host_memory) {
+    // We need to update the amount of physical memory now that
+    // cgroup subsystem files have been processed and has been
+    // determined to be less than the physical host memory.
     os::Linux::set_physical_memory(mem_limit);
     log_info(os, container)("Memory Limit is: " JLONG_FORMAT, mem_limit);
+  } else if (mem_limit >= host_memory) {
```

Comment:
We should probably change this to `else if (log_is_enabled(Trace, os, container) && mem_limit >= host_memory) {...` as we only do trace logs in that branch.

Reply:
Thanks for the catch. Addressed in `7a63709ff7a`